### PR TITLE
Sending Sorted Attendees List and Validation for CloseRollCall

### PR DESCRIPTION
**Pull request by Kaz-ookid**
### Summary

Addressing [issue #1847](https://github.com/dedis/popstellar/issues/1847). FE2 Organizers were sending unordered list of scanned attendees for a `RollCall`, enhancing risks of de-anonymization.

### Main Changes

- Organizer now sends a sorted list of attendees when closing a RollCall.
- UI displays the attendees list in the same order it is received when opening a closed RollCall.
- If the list displayed is not sorted, a toast warns the user about the risk of de-anonymization.
- We do not reorder the list if it is sent unordered, we just warn the user, because if the list has been sent unordered then it is too late already.
- Changed all utilization of `HashSet` to `LinkedHashSet` (in `RollCall` scope) to keep the set ordered as it came in

### Other Changes

I added checking for ordered attendees list at creation, so I also added full validation of message data while I was here.

- Added full validation for the CloseRollCall message data.
- Added complete tests for the CloseRollCall validation.

### UI Updates

- Ensured the attendees list is displayed in the same order it is received.
- Added a toast notification if the list is not sorted.

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragment.kt [hunk 1]
```diff
@@ -342,17 +345,30 @@ class RollCallFragment : AbstractEventFragment {
 
   companion object {
     val TAG: String = RollCallFragment::class.java.simpleName
+    private val deAnonymizationWarned = MutableLiveData(false)
```
Comment (matteosz) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragment.kt:348]: Why putting this as a static constant when the scope it's the Fragment instance?

The method isAttendeeListSorted is called from the RollCallAdapter, to which you can pass either the reference for such live data or the fragment itself.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/RollCallHandler.kt [hunk 2]
```diff
@@ -125,8 +125,24 @@ constructor(
...
+      // We are not ordering it because it is important to keep the order that we received to know
+      // if we face de-anonymization
+      currentAttendees = existingRollCall.attendees
+      currentAttendees.addAll(closeRollCall.attendees)
+    }
+
+    // log existingRollCall.attendees and closeRollCall.attendees
+    Timber.tag(TAG).d("existingRollCall.attendees: %s", existingRollCall.attendees)
+    Timber.tag(TAG).d("closeRollCall.attendees: %s", closeRollCall.attendees)
+    // log currentAttendees
```
Comment (matteosz) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/RollCallHandler.kt:144]: Are these logs something we want to keep or were they there just for debugging purposes?

File: fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragmentTest.kt [hunk 3]
```diff
@@ -488,21 +520,65 @@ class RollCallFragmentTest {
...
+
+    UITestUtils.assertToastDisplayedHasNotText(R.string.roll_call_attendees_list_not_sorted)
+  }
+
+  @Test
+  fun unSortedAttendeeListShowsToast() {
+    rollCallRepo.updateRollCall(LAO_ID, ROLL_CALL_UNSORTED_ATTENDEES)
+    rollCallRepo.updateRollCall(LAO_ID, closeRollCall(ROLL_CALL_UNSORTED_ATTENDEES))
+    InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+    openRollCallWithDescription(ROLL_CALL_UNSORTED_ATTENDEES)
```
Comment (matteosz) [inline fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragmentTest.kt:547]: I think that this call here may actually be useless, it should be sufficient to update the repository (as long as you modify the roll call that is currently set in the fragment)

**Review by matteosz — COMMENTED** [review]
Really good job! Just few comments

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/RollCallHandler.kt (hunk 2 above)
Comment (Kaz-ookid) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/RollCallHandler.kt:144]: good catch, they were just used for debuging

File: fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragmentTest.kt (hunk 3 above)
Comment (Kaz-ookid) [reply fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallFragmentTest.kt:547]: I am not sure, tests do not pass the rollCallListAttendees visibility part if I don't put this line

**Review by matteosz — APPROVED** [review]
LGTM now!

**Pull request merged**